operatorcert/__init__.py

```diff
@@ -311,3 +312,59 @@ def verify_pr_uniqueness(
             for duplicate in duplicate_prs:
                 logging.error(f"DUPLICATE: {duplicate}")
             raise RuntimeError("Multiple pull requests for one Operator Bundle")
+
+
+def download_artifacts(args, resource: str) -> Optional[str]:
+    """
+    Try to get the test-results or artifacts for given parameters from the Pyxis.
+    On success, store the results in files and return True, on failure- return False.
+    """
+
+    if resource not in ["test-results", "artifacts"]:
+        raise ValueError(
+            f"Preflight results are either test-results or artifacts. Got {resource} instead"
+        )
+
+    # If there are multiple results, we only want the most recent one- we enforce it by sorting by creation_date
+    # and getting only the first result.
+    test_results_url = urljoin(
+        args.pyxis_url,
+        f"v1/projects/certification/id/{args.cert_project_id}/{resource}?"
+        f"filter=certification_hash=='{args.certification_hash}';"
+        f"version=='{args.operator_package_version}';"
+        f"operator_package_name=='{args.operator_name}'"
+        f"&sort_by=creation_date[desc]&page_size=1",
+    )
+
+    rsp = requests.get(
+        test_results_url, cert=(args.cert_path, args.key_path), verify=False
+    )
+    rsp.raise_for_status()
+    query_results = rsp.json()["data"]
+
+    if len(query_results) == 0:
+        logging.error(f"There is no {resource} for given parameters")
+        return None
+
+    # Get needed data from the query result
+    # artifacts
+    if resource == "artifacts":
+        result = base64.b64decode(query_results[0]["content"]).decode("utf-8")
+        file_path = "artifact.txt"
```

Comment:
The file name can be gathered from the pyxis artifact response.

Reply:
I'm using the fixed file name to pass the script output to tekton step